### PR TITLE
chore(sdk): improve bucket tests readability with `assertThrows`

### PR DESCRIPTION
**Pull request by garysassano**
Updating the bucket tests following the same standard used in the other resources.

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

**Comment by garysassano** [comment]
@tsuf239 I tried to remove this: https://github.com/winglang/wing/blob/2d1707c3a01ff94cdf7627a325bcb6e325f0cef2/examples/tests/sdk_tests/bucket/public_url.w#L28-L31

But the `http.get(publicUrl).body ` operation throws an error for the `sim` target:
```sh
fail ┌ public_url.wsim » root/env0/test:publicUrl
     │ TypeError: fetch failed
     │     at Object.fetch (node:internal/deps/undici/undici:11457:11)
     │     at async Util.fetch (/tmp/wing-bundles-NFD9aL/index.js:1768:21)
     │     at async $Closure1.handle (/tmp/wing-bundles-NFD9aL/index.js:114:26)
     │     at async exports.handler (/tmp/wing-bundles-NFD9aL/index.js:5713:10)
     │     at async Object.withTrace (/home/user/github/wing/libs/wingsdk/lib/testing/simulator.js:256:34)
     │     at async TestRunnerClient.runTest (/home/user/github/wing/libs/wingsdk/lib/target-sim/test-runner.inflight.js:31:13)
     │     at async testSimulator (/home/user/github/wing/apps/wing/dist/commands/test.js:214:22)
     │     at async testOne (/home/user/github/wing/apps/wing/dist/commands/test.js:125:20)
     │     at async testFile (/home/user/github/wing/apps/wing/dist/commands/test.js:55:39)
     │     at async Promise.all (index 0)
     │     at async Object.test (/home/user/github/wing/apps/wing/dist/commands/test.js:66:5)
     └     at async Command.<anonymous> (/home/user/github/wing/apps/wing/dist/cli.js:46:30)
```

File: examples/tests/sdk_tests/bucket/delete.w [hunk 1]
```diff
@@ -4,7 +4,18 @@ let b = new cloud.Bucket();
 b.addObject("file2.txt", "Bar");
 
 test "delete" {
-  let var error = "";
+  let assertThrows = (expected: str, block: (): void) => {
```
Comment (eladb) [inline examples/tests/sdk_tests/bucket/delete.w:7]: Let's create a `testing` submodule in the Wing SDK and add it there?

```js
bring testing;

testing.assertThrows("message", () => { ... });
```

We can add `testing.assertEquals(x, y)` and `testing.assertNotEquals(x, y)` there as well to improve diagnostics for equality assertions.

@Chriscbr any objections?

Comment (garysassano) [reply examples/tests/sdk_tests/bucket/delete.w:7]: FYI: #3901

**Comment by tsuf239** [comment]
> #2833

Reopened the issue, thanks for reporting @garysassano!

File: examples/tests/sdk_tests/bucket/delete.w (hunk 1 above)
Comment (Chriscbr) [reply examples/tests/sdk_tests/bucket/delete.w:7]: @eladb Sounds good to me. I might suggest `assertEq` or `assertNotEq` just to save some verbosity

Comment (garysassano) [reply examples/tests/sdk_tests/bucket/delete.w:7]: FYI: #3918

**Review by staycoolcall911 — APPROVED** [review]

**Pull request merged**